gym_ignition/base/robot/robot_baseframe.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2019 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+import numpy as np
+from typing import Tuple
+from abc import ABC, abstractmethod
+
+# TODO: new exception for not existing joints? Add Exceptions in the docs?
+
+
+class RobotBaseFrame(ABC):
+    """
+    Robot base frame interface.
+
+    This interface provides methods to get and set quantities related to the robot base.
+    """
+
+    def __init__(self) -> None: ...
+
+    @abstractmethod
+    def set_base_frame(self, frame_name: str) -> bool:
+        """
+        Set the base frame of the robot. If this method is not called, the detection of
+        the base frame depends on the implementation.
```

Comment:
Are you sure this is easy to implement for all possibly backends? I guess for most minimal-coordinates engines it may be not obvious to implement it.

Reply:
>  What do you think?

Ok, this make sense.